component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -170,6 +180,47 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    private String getMatchedSubjectAttribute(String certAttributes, String subjectAttributePattern,
+            AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+
+        LdapName ldapDN;
+        try {
+            ldapDN = new LdapName(certAttributes);
+        } catch (InvalidNameException e) {
+            throw new AuthenticationFailedException("error occurred while get the certificate claims", e);
+        }
+        if (subjectPatternCompiled == null) {
```

Comment:
This is not desired. Pattern is a fixed one scoped for class. Can this be altered per request ?

Reply:
done